rllib/execution/parallel_requests.py

```diff
@@ -236,13 +236,15 @@ def __init__(
         self._ray_wait_timeout_s = ray_wait_timeout_s
         self._return_object_refs = return_object_refs
         self._max_remote_requests_in_flight = max_remote_requests_in_flight_per_worker
-        self._all_workers = set(workers)
         self._pending_to_actor = {}
         self._pending_remotes = []
-        self._worker_queue = OrderedDict.fromkeys(workers)
         self._remote_requests_in_flight = defaultdict(set)
-        self._unavailable_workers = set()
-        self._workers_to_remove = set()
+
+        self._all_workers = (
+            list(workers) if not isinstance(workers, list) else workers.copy()
+        )
+        self._actor_to_ptr = {worker: i for i, worker in enumerate(self._all_workers)}
```

Comment:
I don't think you need this map actually. would make things even simpler.
I will comment on all the use cases of this map below.